assembly/src/parsers/context.rs

```diff
@@ -466,7 +466,7 @@ impl ParserContext {
 
             "sdepth" => simple_instruction(op, Sdepth),
             "locaddr" => io_ops::parse_locaddr(op),
-            "caller" => io_ops::parse_caller(op), // TODO: error if not in SYSCALL
+            "caller" => simple_instruction(op, Caller),
```

Comment:
let's keep this TODO. I'm guessing that's the reason there was still a dedicated `parse_caller`, although I think it's fine to change this in the meantime, since it's currently just duplicated code. Maybe we should also make an issue to track adding this error - there doesn't seem to be one

Reply:
My rationale was that this is so little code that writing it when necessary is not a burden. And using the `simple_instruction` makes it easier to skim over the parser to get its behavior.

I'm fine with either way though :)

Reply:
To be clear - I completely agree with your rationale, I just don't want to lose the inline `TODO` comment

Reply:
Ah! I didn't get it at first. I'm adding it back

Reply:
Updated the issue to #551